src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,154 @@
+import classNames from 'classnames';
+import { useEffect, useRef, useState } from 'react';
+import {
+  CSSTransition,
+  TransitionGroup,
+} from 'react-transition-group';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todos: Todo[];
+  isProcessing: number[];
+  onStatusChange: (todoId: number) => void;
+  onTitleChange: (todoId: number, newTitle: string) => void;
+  onDeleteTodo: (todoId: number) => void;
+};
+
+export const TodoList: React.FC<Props> = (props) => {
+  const {
+    todos,
+    isProcessing,
+    onStatusChange,
+    onTitleChange,
+    onDeleteTodo,
+  } = props;
+  const selectedTodoField = useRef<HTMLInputElement>(null);
+  const [selectedTodo, setSelectedTodo] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState<string>('');
+
+  const handleStartEditTitle = (id: number, title: string) => {
+    setSelectedTodo(id);
+    setNewTitle(title);
+  };
+
+  const handleChangeTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTitle(event.target.value);
+  };
+
+  const handleSaveChages = (todoId: number, title: string) => {
+    setSelectedTodo(null);
+    if (newTitle === '') {
+      onDeleteTodo(todoId);
+    } else if (newTitle !== title) {
+      onTitleChange(todoId, newTitle);
+    }
+  };
+
+  const handleCancelChanges = (
+    event: React.KeyboardEvent<HTMLInputElement>,
+  ) => {
+    if (event.key === 'Escape') {
+      setSelectedTodo(null);
+    }
+  };
+
+  const handleDeleteTodo = (todoId: number) => {
+    onDeleteTodo(todoId);
+  };
+
+  useEffect(() => {
+    if (selectedTodoField.current) {
+      selectedTodoField.current.focus();
+    }
+  }, [selectedTodo]);
+
+  return (
+    <TransitionGroup>
+      {todos.map(todo => {
+        const {
+          id,
+          title,
+          completed,
+        } = todo;
+
+        return (
+          <CSSTransition
```

Comment:
:fire: TransitionGroup- great implementation

Reply:
Thanks, I got fun messing around with it =)